### ISSUE TITLE
how to add empty container element in repsonse message for GET RPC

### ISSUE THREAD

**Issue by NickZengXL** [CESNET/netopeer2#195]
I'm using Netopeer2 0.3.75. I found a problem that when a container is empty, it won't be added into resoponse message for GET RPC.

```
grouping Key-Value-Pair {
    leaf key {
      type Custom-Key;
    }
    leaf value {
      type Custom-Value;
    }
  }

container custom-lists {
      list custom-list {
        key index;

        leaf index {
          type uint8;
        }

        container config {
             list entry {
                key key;
                uses Key-Value-Pair;
            }
         }

      } 

    }
```

In above example yang definition, I have two different reulsts of GET response:
```
  <custom-lists>
      <custom-list>
        <index>1</index>
        <config>
          <entry>
            <key>test</key>
            <value>test1</value>
          </entry>
        </config>
      </custom-list>
      <custom-list>
        <index>2</index>
      </custom-list>
    </custom-lists>
```

You could find that in index 2, there is no `<config>`. Because there is no any entry in index 2.

So, could I change this behavior? I need `<config/>` in index 2 even there is no any entry.

**Comment by michalvasko** [comment]
Hi,
sorry, you cannot strictly change the behaviour and have `<config/>` in the data if there is no list entry. However, you can use the `with-defaults` parameter and specify one of the options that will suit you. We treat non-presence containers exactly as default values because they behave in the same way.

Also, that empty container should have no configuration meaning (as stated [here](https://tools.ietf.org/html/rfc7950#section-7.5.1)), so it would be best for your application to not require this at all.

Regards,
Michal

**Comment by NickZengXL** [comment]
@michalvasko 
Thanks for your reply.

I tried to change second param of nc_server_reply_data in function op_get of file op_get_config.c,
NC_WD_ALL, NC_WD_ALL_TAG,NC_WD_EXPLICIT they all can not add `<config/>` .

And this with-defaults config in function server_init:
/* set with-defaults capability basic-mode */
    nc_server_set_capab_withdefaults(NC_WD_EXPLICIT, NC_WD_ALL | NC_WD_ALL_TAG | NC_WD_TRIM | NC_WD_EXPLICIT);

Could help me about how to change the code to suit my requirement?

**Comment by michalvasko** [comment]
Hi,
how exactly are you working with the response? I was talking about *netopeer2-cli*, to set some `defaults` parameter for the `get` or `get-config` commands, that should work.

Regarding your modifications, the default value flag is a **printing** flag (as specified in `libyang.h`) meaning it should be set on the client side, all the default values (and empty containers) are already there (in the data tree), they are just skipped normally. Also, the function `nc_server_set_capab_withdefaults` simply sets the server capabilities so if you keep this line intact, the server will support all the default modes, no reason to change it.

Regards,
Michal

**Comment by NickZengXL** [comment]
@michalvasko 
I use NETCONc to be netconf client in my test. What I want is NETCONFc can receive empty container.
```
         container config {
             presence "for test";
             list entry {
                key key;
                uses Key-Value-Pair;
            }
         }
```
I also add `presence` statement in my yang definition, and try all four with-defaulf params in get and get-config on NETCONFc.  Still can not see empty containers in response message from netopeer server.

**Comment by michalvasko** [comment]
Hi,
I have tried installing a module with a presence container, I then created the container using `edit-config` with *netopeer2-cli* and after issuing `get-config` (no defaults parameter) I received the container. Meaning I cannot reproduce your problem (I am using `devel` version of *sysrepo*/*netopeer2-server*).

Now, I can help you further only if you provide a use-case with a problem I can easily reproduce with our applications. Otherwise, I can suggest you start *netopeer2-server* with parameter `-c MSG` and it will then print all the messages it sends to a NETCONF client. If you see the container is missing there, post the whole output here and it  may be enough to decide whether there is a bug and how to fix it.

Regards,
Michal
